giraffe/src/utils/fromFlux.ts

```diff
@@ -206,31 +245,57 @@ export const fastFromFlux = (fluxCSV: string): FromFluxResult => {
     let columnKey = ''
     let columnDefault: any = ''
 
-    for (const chunk of chunks) {
-      const splittedChunk = chunk.split('\n')
+    let chunk = ''
+    for (const [start, end] of chunks) {
+      let index = 0
+      chunk = fluxCSV.substring(start, end)
 
-      const tableTexts = []
-      const annotationTexts = []
+      annotationText = ''
+      tableText = ''
 
-      splittedChunk.forEach(line => {
-        if (line.startsWith('#')) {
-          annotationTexts.push(line)
+      while (index !== -1) {
+        const oldIndex = index
+        index = chunk.substring(oldIndex, chunk.length).search(/\n/)
+
+        if (index === -1) {
+          if (chunk[oldIndex] === '#') {
```

Comment:
this is incorrect logic. you should move back to the old logic here and handle it in a different PR. the correct way iirc is to not split lines at all (and let papaparse do that to handle embedded json cases) and look for the first newline that is not followed by a `#`. everything before that point is annotation, every thing after is table text